### PR TITLE
Adding more string operations

### PR DESCRIPTION
**Pull request by brimoor**
I found myself wanting these convenient wrappers around `ViewExpression.re_match()`:

```py
import fiftyone.zoo as foz
from fiftyone import ViewField as F

dataset = foz.load_zoo_dataset("cifar10", split="test")

print(len(dataset.match(F("filepath").starts_with("/Users/Brian")))) # 10000
print(len(dataset.match(F("filepath").starts_with("/USERS/BRIAN")))) # 0
print(len(dataset.match(F("filepath").starts_with("/USERS/BRIAN", case_sensitive=False)))) # 10000

print(len(dataset.match(F("filepath").ends_with("000001.jpg")))) # 1
print(len(dataset.match(F("filepath").ends_with(["000001.jpg", "000002.jpg"])))) # 2

print(len(dataset.match(F("filepath").contains_str("data/00000")))) # 9
print(len(dataset.match(F("filepath").contains_str("DATA/00000")))) # 0
print(len(dataset.match(F("filepath").contains_str("DATA/00000", case_sensitive=False)))) # 9
```

### CODE REVIEW

**Comment by tylerganter** [comment]
Nevermind I'm adding it now

**Comment by tylerganter** [comment]
Yep, tested, works!

```python
import fiftyone as fo
from fiftyone import ViewField as F

# Load the dataset
dataset = fo.Dataset()

dataset.add_samples([
    fo.Sample(filepath="test1.jpg", test_str="lion"),
    fo.Sample(filepath="test2.jpg", test_str="tiger"),
    fo.Sample(filepath="test3.jpg", test_str="bear"),
])

print(len(dataset.match(F("test_str").matches_str("lion"))))  # 1
print(len(dataset.match(F("test_str").matches_str(["lion", "tiger"]))))  # 2

print(len(dataset.match(F("test_str").matches_str("lion_and_more")))) # 0

print(len(dataset.match(F("test_str").matches_str("LION", case_sensitive=True)))) # 0
print(len(dataset.match(F("test_str").matches_str(["LION", "TIGER"], case_sensitive=True)))) # 0

print(len(dataset.match(F("test_str").matches_str("LION", case_sensitive=False)))) # 1
print(len(dataset.match(F("test_str").matches_str(["LION", "TIGER"], case_sensitive=False)))) # 0
```

File: fiftyone/core/expressions.py [hunk 1]
```diff
@@ -502,6 +505,82 @@ def re_match(self, regex, options=None):
...
+
+        Args:
+            str_or_strs: a string or iterable of strings
+            case_sensitive (True): whether to perform a case sensitive match
+
+        Returns:
+            a :class:`ViewExpression`
+        """
+        if etau.is_str(str_or_strs):
+            regex = "^" + str_or_strs + "$"
```
Comment (brimoor) [inline fiftyone/core/expressions.py:577]: @tylerganter FYI regex tokens were missing here

**Comment by brimoor** [comment]
When taking @lethosor up on his request to write unit tests, I realized that the implementations of these methods were not appropriately escaping regex characters. Now that's fixed; EG, one can now do `dataset.match(F("test").contains_str(".")` and it will match the literal `.`.

We also now have B-E-A-U-T-I-L-F-U-L unit tests (not really, but they exist).

**Pull request merged**